src/server/MapPlaylist.ts

```diff
@@ -23,11 +23,32 @@ export class MapPlaylist {
     return mapsPlaylist.shift()!;
   }
 
+  private gameModeHistory: GameMode[] = [];
+
   public getNextGameMode(): GameMode {
-    const nextGameMode = this.gameModeRotation[this.currentGameModeIndex];
-    this.currentGameModeIndex =
-      (this.currentGameModeIndex + 1) % this.gameModeRotation.length;
-    return nextGameMode;
+    const FFA_WEIGHT = 2;
+    const TEAM_WEIGHT = 1;
+    const MAX_HISTORY = 2;
+
+    const last = this.gameModeHistory[this.gameModeHistory.length - 1];
+    const secondLast = this.gameModeHistory[this.gameModeHistory.length - 2];
+
+    // Avoid repeating the same game mode 3 times in a row
+    if (last === secondLast && last !== undefined) {
+      const opposite = last === GameMode.FFA ? GameMode.Team : GameMode.FFA;
+      this.gameModeHistory.push(opposite);
+      if (this.gameModeHistory.length > MAX_HISTORY)
+        this.gameModeHistory.shift();
+      return opposite;
+    }
+
+    const roll = Math.floor(Math.random() * (FFA_WEIGHT + TEAM_WEIGHT));
```

Comment:
floor has no effect on the outcome
```suggestion
    const roll = Math.random() * (FFA_WEIGHT + TEAM_WEIGHT);
```

Reply:
Hey, thanks for the feedback! 

afaik this has already been implemented in a better way in v22.